eclipse-che/che-server
# chore: Remove some outdated PVC storage properties

### What does this PR do?
Removes the following Che-Server properties (alongside some classes related to these properties), as PVC configuration in Eclipse Che is now managed by the devworkspace-operator:
```
che.workspace.projects.storage.default.size
che.infra.kubernetes.pvc.enabled
che.infra.kubernetes.pvc.strategy
che.infra.kubernetes.pvc.precreate_subpaths
che.infra.kubernetes.pvc.name
che.infra.kubernetes.pvc.storage_class_name
che.infra.kubernetes.pvc.quantity
che.infra.kubernetes.pvc.jobs.image
che.infra.kubernetes.pvc.jobs.image.pull_policy
che.infra.kubernetes.pvc.jobs.memorylimit
che.infra.kubernetes.pvc.access_mode
che.infra.kubernetes.pvc.wait_bound
```

### Screenshot/screencast of this PR
n\a

### What issues does this PR fix or reference?
Part of eclipse/che#21405

### How to test this PR?
While working on this PR, the workflow to test my changes was the following:
**1.** Build and test relevant modules, ensure no errors:
```bash
mvn clean install -DskipTests=false -Dfindbugs.skip=true -Dmdep.analyze.skip=true -Dlicense.skip=true -Dgwt.compiler.localWorkers=2 -T 1C -pl ':infrastructure-kubernetes,:infrastructure-openshift' --amd -Pintegration
```
**2.** Do a full build with tests, ensure no errors
`mvn clean install -Pintegration`
**3.** **Slightly outside [the scope](https://github.com/eclipse-che/che-operator/pull/1442#issuecomment-1184638247) of this PR:** Ensure it is possible to deploy che-server with chectl with no errors, while using an updated version of the che-cluster CR, that uses the 'per-user' workspace storage strategy.
**a.** Clone my fork of che-operator: `git clone git@github.com:eclipse-che/che-operator.git && cd che-operator && git checkout -b AObuchow-update_che_cr_storage_api_2 && git pull git@github.com:AObuchow/che-operator.git update_che_cr_storage_api_2`
**b.** Generate CR template from within the che-operator repo, on my branch `make gen-chectl-tmpl TARGET=/tmp/operator-resources`
**c.** Deploy che-server with my che-server image (built from the current PR) and my che-operator image (built from [this PR](https://github.com/eclipse-che/che-operator/pull/1442)) as well as the updated che-cluster CR:
`chectl server:deploy --installer operator -p minikube -i=quay.io/aobuchow/che-server:next --che-operator-image=quay.io/aobuchow/che-operator:next --debug --templates /tmp/operator-resources/che-operator/`

Alternatively, you can build your own che-operator and che-server images and host them on your own quay repository. If needed, I can post instructions on how to do so.

**d.** Ensure che-server does not contain any errors in its logs:
`chectl server:logs -d ~/tmp/ && cd ~/tmp/eclipse-che/ && cd $(find . -name che.log -exec dirname {} + ) && cat che.log`


### Additional notes:
- Unfortunately, the commits are not as cleanly divided as they could be. Most commits are named after the properties removed from che.properties, however classes relevant to that property may only be removed in a later commit. If this complicates the review process too much, I can rework this PR so that the classes removed in each commit are only relevant to the property being removed.
- There are more PVC related functions and classes that could be removed, but it would make this PR much bigger than it already is. It is probably best to leave those for another PR.
- [Che-Docs will likely also require a PR](https://github.com/eclipse-che/che-docs/blob/main/modules/administration-guide/examples/system-variables.adoc#che_workspace_projects_storage_default_size) to remove the mention of the properties that is PR removes.



### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
